src/python/pants/engine/target.py

```diff
@@ -272,12 +287,28 @@ def get(self, field: Type[_F]) -> _F:
             return cast(_F, self.field_values[field_subclass])
         raise KeyError(
             f"The target `{self}` does not have a field `{field}`. Before calling "
-            f"`my_tgt.get({field.__name__})`, call `my_tgt.has_fields([{field.__name__}])` to "
+            f"`my_tgt.get({field.__name__})`, call `my_tgt.has_field({field.__name__})` to "
             "filter out any irrelevant Targets."
         )
 
+    @final
+    def has_field(self, field: Type[Field]) -> bool:
+        """Check that this target has registered the requested field.
+
+        This works with subclasses of `Field`s. For example, if you subclass `Sources` to define a
+        custom subclass `PythonSources`, both `python_tgt.has_field(PythonSources)` and
+        `python_tgt.has_field(Sources)` will return True.
+        """
+        return self.has_fields([field])
```

Comment:
It might be a micro optimization, but this method could probably have its own implementation to avoid the extra collection classes that are created here.

Reply:
Hm, I looked at factoring that out and it wasn't very pretty. I think this is likely a micro optimization. 

We'll probably want to profile the Target API once the dust settles, and maybe this ends up making a difference to change down the road.